airflow/www/decorators.py

```diff
@@ -53,7 +54,10 @@ def wrapper(*args, **kwargs):
             )
 
             if 'execution_date' in request.values:
-                log.execution_date = pendulum.parse(request.values.get('execution_date'), strict=False)
+                try:
+                    log.execution_date = pendulum.parse(request.values.get('execution_date'), strict=False)
+                except ParserError:
+                    pass
```

Comment:
This silently discard an invalid `execution_date` value and I don’t think it’s a good thing. Why is this needed?

Reply:
If we can't parse the execution date, we can't store it in the execution date column since it is typed. It should still be preserved in the extras though: `extra=str([(k, v) for k, v in request.values.items() if k not in fields_skip_logging]),`.

Reply:
We can’t store it, but the code shouldn’t just ignore it *silently*. This should be bubbled back to wherever the invalid value was generated (presumably from user input?) and show a correct error message. For example, if the value was from a web request, the request should fail with a 400 Bad Request or something like that, with a clear error message indicating what went wrong.

Reply:
It is being flashed as an error on the location it happens: https://github.com/apache/airflow/pull/16141/files#diff-917b70d2661fae8322c538a4f60b9d4ba755556036fe0c527eb0411a6ac60f2eR1517
This is a decorator for user activity logging as far as I understand. Doesn't seem like it should be its concern to validate and surface user input issues.

Reply:
Then maybe log a warning that the `execution_date` value is being dropped? That would help debug why that happens when reading the log (since the UI flash would likely be gone by the point logs are accessed).

Reply:
Sounds good.